aspnetcore/security/authorization/secure-data/samples/final3/Startup.cs

```diff
@@ -42,17 +41,11 @@ public void ConfigureServices(IServiceCollection services)
                 .AddEntityFrameworkStores<ApplicationDbContext>();
             #endregion
 
-            services.AddRazorPages();
-
-            services.AddControllers(config =>
```

Comment:
`[AllowAnonymous]` didn't work for the new `Details2` page, which brought me here to see what was going on. The sample app doesn't have any controllers, so I think it's better to drop this in favour of using the RP convention shown. That also fixes the `[AllowAnonymous]` thing.